### PR TITLE
build: follow up to #5449. implementing suggestions and deduplication 

### PR DESCRIPTION
**Pull request by kittywhiskers**
## Additional Information

* Based on suggestions by @knst made [here](https://github.com/dashpay/dash/pull/5449#issuecomment-1609937147) and [here](https://github.com/dashpay/dash/pull/5426#discussion_r1241789033)

### CODE REVIEW

File: contrib/containers/guix/scripts/guix-check [hunk 1]
```diff
@@ -2,16 +2,27 @@
...
-SRC_PATH_PREFIX=guix-build-${COMMIT_ID}/distsrc- && \
+SRC_PATH_PREFIX="guix-build-${COMMIT_ID}/distsrc-" && \
 sha256sum ${SRC_PATH_PREFIX}*/src/dash{d,-cli,-tx,-wallet}{,.exe} && \
 sha256sum ${SRC_PATH_PREFIX}*/src/qt/dash-qt{,.exe} && \
 sha256sum ${SRC_PATH_PREFIX}*/src/test/test_dash{,.exe} \
 ) | sort -k 2
 
-printf "Archives:\n"
-find guix-build-"${COMMIT_ID}"/output -type f | grep -v SHA256 | xargs sha256sum | sort -k 2
+printf "Archives:\n\n"
```
Comment (UdjinM6) [inline contrib/containers/guix/scripts/guix-check:27]: same
```suggestion
printf "\nArchives:\n\n"
```
```
633b1ef9c3da166f242b47e467347736a8c6052042eae04ad71b123250a70b4e  guix-build-6c6ca52ab1d6/distsrc-6c6ca52ab1d6-x86_64-w64-mingw32/src/test/test_dash.exe

Archives:

1ff0c6025cc789ddf72b1e8b236b9d49245ac854806d46b7abd[43](https://github.com/dashpay/dash/actions/runs/5400271755/jobs/9808483734?pr=5464#step:9:44)56977590fae  guix-build-6c6ca52ab1d6/output/aarch64-linux-gnu/dashcore-6c6ca52ab1d6-aarch64-linux-gnu-debug.tar.gz
```
vs
```
633b1ef9c3da166f242b47e467347736a8c6052042eae04ad71b123250a70b4e  guix-build-6c6ca52ab1d6/distsrc-6c6ca52ab1d6-x86_64-w64-mingw32/src/test/test_dash.exe
Archives:

1ff0c6025cc789ddf72b1e8b236b9d49245ac854806d46b7abd[43](https://github.com/dashpay/dash/actions/runs/5400271755/jobs/9808483734?pr=5464#step:9:44)56977590fae  guix-build-6c6ca52ab1d6/output/aarch64-linux-gnu/dashcore-6c6ca52ab1d6-aarch64-linux-gnu-debug.tar.gz
```

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

File: contrib/containers/guix/scripts/guix-start [hunk 2]
```diff
@@ -2,19 +2,30 @@
...
-    curl -L https://bitcoincore.org/depends-sources/sdks/${XCODE_ARCHIVE}.tar.gz | tar -xz -C /src/dash/depends/SDKs
+    mkdir -p "${WORKSPACE_PATH}/depends/SDKs"
+    curl -L https://bitcoincore.org/depends-sources/sdks/${XCODE_ARCHIVE}.tar.gz | tar -xz -C "${WORKSPACE_PATH}/depends/SDKs"
 fi
 
-cd /src/dash
+cd ${WORKSPACE_PATH}
 git status >> /dev/null
-git config --global --add safe.directory /src/dash
+git config --global --add safe.directory ${WORKSPACE_PATH}
```
Comment (PastaPastaPasta) [inline contrib/containers/guix/scripts/guix-start:29]: (for some reason gh isn't letting me put a suggestion, but can you put the safe.directory above the git status? I needed that diff in order to make it work. Probably should've been fixed earlier

File: contrib/containers/guix/scripts/guix-check (hunk 1 above)
Comment (kittywhiskers) [reply contrib/containers/guix/scripts/guix-check]: Resolved in latest push

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**